### ISSUE TITLE
Training problem in SiamFC with another dataset

### ISSUE THREAD

**Issue by haoliyoupai09** [huanglianghua/open-vot#4]
Hi, I tried to retrain SiamFC in another dataset. But I met a very strange problem. In the manager_siamfc.py file, there are codes as follows:
        # training dataset
        dataset_train = Pairwise(
            base_dataset, transform, subset='train')
        dataloader_train = DataLoader(
            dataset_train, batch_size=self.cfg.batch_size, shuffle=True,
            pin_memory=self.cuda, drop_last=True, num_workers=cpu_num)

        # validation dataset
        dataset_val = Pairwise(
            base_dataset, transform, subset='val')
        dataloader_val = DataLoader(
            dataset_val, batch_size=self.cfg.batch_size, shuffle=False,
            pin_memory=self.cuda, drop_last=True, num_workers=cpu_num)

        train_iters = len(dataloader_train)
        val_iters = len(dataloader_val)
After I changed the training dataset, I run the siamfc.py and found that the value for train_iters  is non-zero, but the value for val_iters is zero. I think the loading process for the train and val data are similar. I don't know what's the meaning of the returned value of len(dataloader_train) and what's going on in my code? 
And when I changed the train_rotio (0.95) in pairwise.py to a smaller value (0.5), the val_iters turned to 1, and we can begin training successfully.
Could you tell us the possibile reason behind this? Thank you very much.

**Comment by huanglianghua** [comment]
@haoliyoupai09 Hi, how large is your new dataset? I suppose this issue is due to the small size of the dataset while the batch size is relatively large. For example, for a batch_size=32 and the size of base_dataset is around 100, the value of val_iters=(len(base_dataset) * (1 - train_ratio)) // batch_size would be small as well.

**Comment by haoliyoupai09** [comment]
@huanglianghua Hi, I think you are right. The size of my dataset is only 97. Thank you very much  for your response.
